src/pudl/workspace/datastore.py

```diff
@@ -302,10 +302,10 @@ def __init__(
                 self._cache.add_cache_layer(
                     resource_cache.GoogleCloudStorageCache(gcs_cache_path)
                 )
-            except DefaultCredentialsError:
+            except (DefaultCredentialsError, OSError) as e:
```

Comment:
Why are you catching this OSError?

Reply:
If the gcloud project isn't passed in, `storage.Client()` throws an OSError instead of a DefaultCredentialsError - but it's still a gcloud config problem. Unfortunately we hard-code a GCS cache path in `pudl.metadata.classes.DataSource.add_datastore_metadata`, so even the little "if you didn't log in to Google then don't try to use Google" workaround above doesn't work. I didn't want to mess with passing more CLI args all the way into `DataSource` so I thought catching this error seemed like a reasonable way to deal with this gcloud config problem.